lib/t/t_pf_radial.m

```
@@ -0,0 +1,138 @@
+    for j = 1:size(method,1)
+        mpopt1 = mpoption(mpopt,'pf.alg',method{j,1});
+        r1 = runpf(mpc,mpopt1);
+        iter_nopv(i,j+1) = r1.iterations;
+        t = [method{j,2} ', ' casefile{i} ' : '];
+        t_ok(r1.success, [t 'success']);
+        t_is(r1.bus, r.bus, 6, [t 'bus']);
+        t_is(r1.gen, r.gen, 6, [t 'gen']);
+        t_is(r1.branch, r.branch, 6, [t 'branch']);
+    end
+end
+
+% Test cases with added PV buses
+iter_pv = zeros(size(casefile,1),size(method,1)+1,2);
+for i = 1:length(casefile)
+    mpc = loadcase(casefile{i});
+    pv = mpc.bus(:,BUS_TYPE) == PV;
+    if all(~pv)
+        %%% ADD PV BUSES %%%
+        N = 5; % How many PV buses should I add?
+        % Solve the case without PV buses with Newton method
+        r = runpf(mpc, mpopt);
+        % Find the last N buses with lowest voltage
+        [Vm, B] = sort(r.bus(:,VM));
+         B = B(1:N);
+        Vm = Vm(1:N);
+        % add PV generators at buses in B
+        % set VG 0.05 pu bigger then voltage at buses in B
+        mpc.gen = repmat(mpc.gen,N+1,1);
+        mpc.gen(2:end,GEN_BUS) = B;
+        mpc.gen(2:end,VG) = Vm + 0.05;
+        mpc.bus(B,BUS_TYPE) = 2;
+        if isfield(mpc,'gencost')
+            mpc.gencost = repmat(mpc.gencost,N+1,1);
+        end
+        %%% END OF ADD PV BUSES %%%
+    end
+    r = runpf(mpc, mpopt);
+    iter_pv(i,1,1) = r.iterations;
+    iter_pv(i,1,2) = r.iterations;
+    for j = 1:size(method,1)
+        for vcorr = 0:1
+            mpopt1 = mpoption(mpopt,'pf.alg',method{j,1},'pf.radial.vcorr',vcorr);
+            r1 = runpf(mpc,mpopt1);
+            iter_pv(i,j+1,vcorr+1) = r1.iterations;
+            t = [method{j,2} ', vcorr = ' num2str(vcorr) ', ' casefile{i} ' : '];
+            t_ok(r1.success, [t 'success']);
+            t_is(r1.bus, r.bus, 6, [t 'bus']);
+            t_is(r1.gen, r.gen, 6, [t 'gen']);
+            t_is(r1.branch, r.branch, 6, [t 'branch']);
+        end
+    end
+end
+
+t_end;
+
```

The printing below should only happen if `verbose` is true. Also, my idea was to take the results of this and hard code it in a cell array, then include a `t_is` that checks `r.iterations` against the corresponding hard-coded value for each run. That way if someone changes the code in a way that causes the algorithm to behave differently, the iteration counts may change, and tests may fail. The goal is to have inadvertent changes show up as failed tests.